Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -0,0 +1,302 @@
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(0, &GradientBakerConfig::m_gradientSampler, "Gradient", "Input gradient to bake the output image from.")
+                    ->DataElement(0, &GradientBakerConfig::m_inputBounds, "Input Bounds", "Input bounds for where to sample the data.")
+                    ;
+            }
+        }
+    }
+
+    void EditorGradientBakerComponent::Reflect(AZ::ReflectContext* context)
+    {
+        GradientBakerConfig::Reflect(context);
+
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EditorGradientBakerComponent, EditorComponentBase>()
+                ->Version(0)
+                ->Field("Configuration", &EditorGradientBakerComponent::m_configuration)
+                ->Field("PreviewEntity", &EditorGradientBakerComponent::m_previewEntityId)
+                ->Field("PreviewPosition", &EditorGradientBakerComponent::m_previewPosition)
+                ->Field("PreviewSize", &EditorGradientBakerComponent::m_previewSize)
+                ->Field("ConstrainToShape", &EditorGradientBakerComponent::m_constrainToShape)
+                ;
+
+            if (auto editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<EditorGradientBakerComponent>(
+                    EditorGradientBakerComponent::s_componentName, EditorGradientBakerComponent::s_componentDescription)
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Icon, s_icon)
+                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, s_viewportIcon)
+                    ->Attribute(AZ::Edit::Attributes::HelpPageURL, s_helpUrl)
+                    ->Attribute(AZ::Edit::Attributes::Category, EditorGradientBakerComponent::s_categoryName)
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game", 0x232b318c))
+                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+
+                    ->DataElement(0, &EditorGradientBakerComponent::m_configuration, "Configuration", "")
+                    ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorGradientBakerComponent::OnConfigurationChanged)
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
+
+                    ->ClassElement(AZ::Edit::ClassElements::Group, "Preview")
+                    ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::Show)
+                    ->UIElement(AZ_CRC("GradientPreviewer", 0x1dbbba45), "Previewer")
+                    ->Attribute(AZ::Edit::Attributes::NameLabelOverride, "")
+                    ->Attribute(AZ_CRC("GradientEntity", 0xe8531817), &EditorGradientBakerComponent::GetGradientEntityId)
```

Comment:
Suggestion: You can use AZ_CRC_CE instead of AZ_CRC and not need to put in the calculated CRC values.

Reply:
Will update!